src/deal_ii/src/4C_deal_ii_element_conversion.hpp

```diff
@@ -10,213 +10,492 @@
 
 #include "4C_config.hpp"
 
-#include "4C_fem_discretization.hpp"
 #include "4C_fem_general_element.hpp"
 #include "4C_fem_general_node.hpp"
+#include "cut/4C_cut_find_cycles.hpp"
 
 #include <deal.II/fe/fe_simplex_p.h>
 #include <deal.II/fe/fe_system.h>
 #include <deal.II/fe/fe_tools.h>
-#include <deal.II/hp/fe_collection.h>
 
 FOUR_C_NAMESPACE_OPEN
 
-namespace DealiiWrappers::ElementConversion
+
+
+namespace DealiiWrappers
 {
+  namespace ConversionTools
+  {
+    namespace Internal
+    {
+      template <bool index_runs_over_four_c, int dim, int spacedim = dim>
+      void reindex_shape_functions(Core::FE::CellType, unsigned int n_components,
+          const std::span<const int>& reindex_scalar, std::vector<int>& reindexing);
+    }  // namespace Internal
+
+    namespace FourCToDeal
```

Comment:
Maybe the two namespaces are a bit much. How about just `FourCToDealii` and remove `ConversionTools`?